amethyst_ui/src/button/builder.rs

```diff
@@ -39,16 +40,16 @@ pub struct UiButtonBuilderResources<'a, G: PartialEq + Send + Sync + 'static> {
     parent: WriteStorage<'a, Parent>,
     text: WriteStorage<'a, UiText>,
     transform: WriteStorage<'a, UiTransform>,
-    button: WriteStorage<'a, UiButton>,
+    button_widgets: WriteExpect<'a, Widgets<UiButton, I>>,
     sound_retrigger: WriteStorage<'a, UiSoundRetrigger>,
     button_action_retrigger: WriteStorage<'a, UiButtonActionRetrigger>,
     selectables: WriteStorage<'a, Selectable<G>>,
 }
 
 /// Convenience structure for building a button
 #[derive(Debug, Clone)]
-pub struct UiButtonBuilder<G> {
-    name: String,
+pub struct UiButtonBuilder<G, I: WidgetId> {
```

Comment:
I thought the point of using predefined implementations with strings and u32 was to avoid adding a bound like this on the builder by hardcoding a u32. Is it just for automatic generation in this case?

Reply:
Yes, this makes sure that when the user doesn't provide any ids at all in his prefab (also important for backwards compat), he will automatically get `u32`s and can still iterate over the `Widgets` resource.